### PR TITLE
app: add new shell_overlay.conf

### PR DESCRIPTION
**Pull request by kv2019i**
Add overlay to build SOF with Zephyr shell enabled in the build.

Depends on https://github.com/zephyrproject-rtos/zephyr/pull/72738

### CODE REVIEW

**Comment by kv2019i** [comment]
Keeping as draft until Zephyr part has been merged.

**Comment by kv2019i** [comment]
Dependencies now merged, rebased to include the dependencies and marking ready for review.

Documentation at https://github.com/thesofproject/sof-docs/pull/496 (also waiting for reviews).

**Review by jsarha — APPROVED** [review]
The overlay works. I guess that's good enoug.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**
